lightning/src/chain/channelmonitor.rs

```diff
@@ -131,7 +131,40 @@ pub enum MonitorEvent {
 
 	/// A monitor event that the Channel's commitment transaction was confirmed.
 	CommitmentTxConfirmed(OutPoint),
+
+	/// Indicates a [`ChannelMonitor`] update has completed. See
+	/// [`ChannelMonitorUpdateErr::TemporaryFailure`] for more information on how this is used.
+	///
+	/// [`ChannelMonitorUpdateErr::TemporaryFailure`]: super::ChannelMonitorUpdateErr::TemporaryFailure
+	UpdateCompleted {
+		/// The funding outpoint of the [`ChannelMonitor`] that was updated
+		funding_txo: OutPoint,
+		/// The Update ID from [`ChannelMonitorUpdate::update_id`] which was applied or
+		/// [`ChannelMonitor::get_latest_update_id`].
+		///
+		/// Note that this should only be set to a given update's ID if all previous updates for the
+		/// same [`ChannelMonitor`] have been applied and persisted.
+		monitor_update_id: u64,
+	},
+
+	/// Indicates a [`ChannelMonitor`] update has failed. See
+	/// [`ChannelMonitorUpdateErr::PermanentFailure`] for more information on how this is used.
+	///
+	/// [`ChannelMonitorUpdateErr::PermanentFailure`]: super::ChannelMonitorUpdateErr::PermanentFailure
+	UpdateFailed(OutPoint),
```

Comment:
Given we may not prioritize renaming `TemporaryFailure`, could we name this `UpdatePermFailed`? I keep thinking it refers to either...

Reply:
For future naming, I'm wondering if we either use an enum:

```rust
pub enum PersistenceResult { Completed, Pending, Failed }
```

Or a similar enum wrapped by a result:

```rust
pub enum PersistenceStatus{ Completed, Pending }
pub type PersistenceResult = Result<PersistenceStatus, ()>;
```

Reply:
> Given we may not prioritize renaming TemporaryFailure, could we name this UpdatePermFailed?

Lets just prioritize renaming it? I can do it in a followup immediately.